### ISSUE TITLE
不同数据集的维度问题

### ISSUE THREAD

**Issue by ztoString** [Wovchena/text-detection-fots.pytorch#5]
我用个icpr2018数据集，数据格式也完全转换成了ICDAR的样子，但是还是会报错：
File "train.py", line 317, in <module>
    fit(epoch, model, detection_loss, optimizer, lr_scheduler, best_score, args.batches_before_train, checkoint_dir, dl, None)
  File "train.py", line 266, in fit
    loss = loss_func(prediction, (classification, regression, thetas, training_mask)) / max_batches_per_iter_cnt
  File "train.py", line 97, in detection_loss
    raw_cls_loss = torch.nn.functional.binary_cross_entropy_with_logits(input=y_pred_cls, target=y_true_cls, weight=None, reduction='none')
  File "/data/zhengjunjun/wangyan02/anaconda3/envs/pytorch/lib/python3.6/site-packages/torch/nn/functional.py", line 2161, in binary_cross_entropy_with_logits
    raise ValueError("Target size ({}) must be the same as input size ({})".format(target.size(), input.size()))
ValueError: Target size (torch.Size([8, 1, 320, 320])) must be the same as input size (torch.Size([8, 1, 160, 160]))

**Comment by Wovchena** [comment]
@ztoString, thank you for the report. You can try replacing `IN_OUT_RATIO = 2` with `IN_OUT_RATIO = 4` in `datasets.py` and `modules/parse_polys.py`.

**Comment by ghost** [comment]
@Wovchena I got the same error, changing the ratio didnot solve the issue.

**Comment by Wovchena** [comment]
Changing the ratio does the trick for me. Are you running `master` branch? Please, specify the command you are using to run the script.

**Comment by ghost** [comment]
@Wovchena Can you test training with this sample [icdar15.zip](https://github.com/Wovchena/text-detection-fots.pytorch/files/3617255/icdar15.zip)
I am using the master branch.

**Comment by Wovchena** [comment]
It runs for me with 2->4 replacement for ratio:
```
python3 train.py --train-folder /mnt/h/tmp --num-workers 3 --batch-size 5 --batches-before-train 1
Epoch 0: 100%|███████████████████| 2/2 [00:01<00:00,  1.20it/s, Mean loss=3.345]
Updated best_model
Epoch 1: 100%|███████████████████| 2/2 [00:01<00:00,  1.15it/s, Mean loss=3.183]
Updated best_model
```

**Comment by ghost** [comment]
@Wovchena 
- Did you test using the latest master branch?

**Comment by Wovchena** [comment]
Yes, I got this results for `master` d9c727a.
> python3 train.py --train-folder /mnt/h/tmp --num-workers 3 --batch-size 5 --batches-before-train 1
Epoch 0: 100%|███████████████████| 2/2 [00:01<00:00,  1.20it/s, Mean loss=3.345]
Updated best_model
Epoch 1: 100%|███████████████████| 2/2 [00:01<00:00,  1.15it/s, Mean loss=3.183]
Updated best_model

**Comment by liujiale1234** [comment]
img = cv2.imread(os.path.join(os.path.join(self.root, self.img_dir), self.image_prefix[idx] + '.jpg'), cv2.IMREAD_COLOR).astype(np.float32)
MemoryError

How can I solve this problem?

**Comment by Wovchena** [comment]
> img = cv2.imread(os.path.join(os.path.join(self.root, self.img_dir), self.image_prefix[idx] + '.jpg'), cv2.IMREAD_COLOR).astype(np.float32)
> MemoryError
> 
> How can I solve this problem?

I can't say for sure, please, provide full log. And try splitting the line: the error could be in `imread` or in `astype`, may be it will clarify the problem. My guesses are you ran out of memory or the image is corrupted/missed.

**Comment by Wovchena** [comment]
The initial problem is fixed by 816fc8b201ea35be83f57457f42e23067e65a22f.
